src/helpers/referal.js

```diff
@@ -136,14 +169,20 @@ module.exports.api = function (app) {
         const hierarchy = [];
 
         let referList = [],
-            level = 1,
+            level = req.body.level,
             index = 0;
 
+        let limit = 1;
+
+        if (level > 1) {
+            level++;
+        }
+
         function findSponsors(arr, cb) {
-            if (level <= 15) {
+            if (limit <= 5 && level <= 15) {
```

Comment:
strange logic

Reply:
Explain it

Reply:
You can use with recursive

Reply:
Actually due to the logic by which we are fetching the referral list and the recursion process. we can't put the limit or offset based query.